### ISSUE TITLE
Incorrect upper bound on type parameter with inner classes

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#123]
Consider this class,

```java
import java.util.*;
import org.checkerframework.checker.determinism.qual.*;

public class TestTypeParams<T extends @NonDet Object> {
    public class C {
        public C(T a) {}

        public void f(T a) {
            new C(a);
        }
    }
}
```

The constructor for `C` expects a `T`, which is exactly what is passed in `f`. These should both refer to the same type, `T extends @NonDet Object`. When I run the checker I get this error.
```
estTypeParams.java:9: error: [argument.type.incompatible] incompatible types in argument.
            new C(a);
                  ^
  found   : T extends @NonDet Object
  required: T extends @Det Object
```

The upper bound on `T` is different, which is incorrect because they should be referring to the same type.

**Comment by t-rasmud** [comment]
Same as #2865

**Comment by t-rasmud** [comment]
Fixed in 84ba3288a5dcd2cf93bb759813846e316d94edfe
